spec/datadog/tracing/trace_operation_spec.rb

```
@@ -1023,6 +1023,54 @@
     it_behaves_like 'root span derived attribute', :resource
   end
 
+  describe '#set_distributed_source' do
+    context 'when the trace source is not set' do
+      context 'with product bit set to 2' do
+        before do
+          trace_op.set_distributed_source(2)
+        end
+
+        it 'sets the trace source to the product bit' do
+          expect(trace_op.get_tag('_dd.p.ts')).to eq('02')
+        end
+      end
+
+      context 'with product bit set to 16' do
+        before do
+          trace_op.set_distributed_source(16)
+        end
+
+        it 'sets the trace source to the product bit' do
+          expect(trace_op.get_tag('_dd.p.ts')).to eq('10')
+        end
+      end
+
+      context 'with product bit higher than 8th bit' do
+        before do
+          trace_op.set_distributed_source(256)
+        end
+
+        # We must support up to 32 bits for future usage.
+        it 'sets the trace source to the product bit' do
+          expect(trace_op.get_tag('_dd.p.ts')).to eq('100')
+        end
```

is this the expected behaviour that we truncate the tag? Should we raise an exception instead?

---

I'm not sure I understand, I don't think that we don't truncate it, 256 is 100 in hexadecimal. We don't do anything if the value is larger than FF because we must accept at least 32 bits to ensure forward compatibility.